src/Raven.Server/RavenServer.cs

```diff
@@ -1006,18 +1006,17 @@ public bool RefreshClusterCertificate(object state, string raftRequestId)
             var forceRenew = state as bool? ?? false;
 
             var currentRefreshTask = _currentRefreshTask;
-            if (currentRefreshTask.IsCompleted == false)
+            if (currentRefreshTask.Unwrap().IsCompleted == false)
             {
                 _refreshClusterCertificate?.Change(TimeSpan.FromMinutes(1), TimeSpan.FromHours(1));
                 return false;
             }
 
-            var refreshCertificate = new Task(async () => { await DoActualCertificateRefresh(currentCertificate, raftRequestId, forceRenew: forceRenew); });
-            if (Interlocked.CompareExchange(ref _currentRefreshTask, currentRefreshTask, refreshCertificate) != currentRefreshTask)
+            var refreshCertificate = new Task<Task>(() => DoActualCertificateRefresh(currentCertificate, raftRequestId, forceRenew: forceRenew));
```

Comment:
wouldn't using `Lazy<Task>` make it simpler? No unwrapping, etc?

Reply:
good point, will do it with `Lazy<Task>`